### PR TITLE
[FIX] Fix and update Softmax regression learner

### PR DESCRIPTION
**Pull request by PrimozGodec**
##### Issue

Fixes https://github.com/biolab/orange3/issues/4715

##### Description of changes
- Updated softmax regression
- Updated base model in parts where it is required for softmax regression to work correctly.

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Orange/classification/base_classification.py [hunk 1]
```diff
@@ -18,27 +16,8 @@ class ModelClassification(Model):
...
-                probs_ext[:, c, cv] = probs[:, c, i]
-        if not self.supports_multiclass:
-            probs_ext = probs_ext[:, 0, :]
-        return values, probs_ext
+    pass
 
 
 class SklLearnerClassification(SklLearner, LearnerClassification):
-    __returns__ = SklModelClassification
+    __returns__ = SklModel
```
Comment (lanzagar) [inline Orange/classification/base_classification.py:23]: Why was this changed?

File: Orange/base.py [hunk 2]
```diff
@@ -383,6 +380,30 @@ def one_hot_probs(value):
                 probs[:, i, :] = one_hot(value[:, i])
             return probs
 
+        def extend_probabilities(probs):
+            """
+            Since SKLmodels and models with fit function does not guarantee
```
Comment (lanzagar) [inline Orange/base.py:385]:
```
Since SklModels and models implementing `fit` and not `fit_storage` do not guarantee
```

File: Orange/base.py [hunk 3]
```diff
@@ -383,6 +380,30 @@ def one_hot_probs(value):
...
+            correct prediction dimensionality, extend dimensionality of
+            probabilities when it does not mach to the number of values in
+            domain.
```
Comment (lanzagar) [inline Orange/base.py:388]: mach to the number of values in domain -> match the number of values in the domain

File: Orange/classification/base_classification.py (hunk 1 above)
Comment (PrimozGodec) [reply Orange/classification/base_classification.py]: It is my mistake. First I wanted to get rid of SklLearnerClassification class, but then I decided to do refactoring in separate (following) PR. It would be too many changes for this PR.
